lib/match.js

```diff
@@ -3,13 +3,10 @@
 var valueToString = require("@sinonjs/commons").valueToString;
 var indexOf = require("@sinonjs/commons").prototypes.string.indexOf;
 var forEach = require("@sinonjs/commons").prototypes.array.forEach;
+var type = require("type-detect");
 
 var engineCanCompareMaps = typeof Array.from === "function";
 var deepEqual = require("./deep-equal").use(match); // eslint-disable-line no-use-before-define
-var getClass = require("./get-class");
-var isDate = require("./is-date");
-var isSet = require("./is-set");
-var isMap = require("./is-map");
```

Comment:
Doesn't this mean that our own type detection and the one from `type-detect` could theoretically differ? Maybe we should use `type-detect` in these helpers as well 🤔

Reply:
I was thinking the same thing. I don't really know if we need these helpers anymore. If we decide to deprecate them, then let's give it a couple of major versions before we remove them

Reply:
I wouldn't deprecate them. They can be simply forward to `type-detect`. It's nice to have all our type detection and object comparison needs in a single package.